packages/autorest.typescript/test/integration/generated/appconfigurationexport/src/appConfigurationClient.ts

```diff
@@ -152,22 +153,31 @@ export class AppConfigurationClient extends coreClient.ServiceClient {
       [Symbol.asyncIterator]() {
         return this;
       },
-      byPage: () => {
-        return this.getKeysPagingPage(options);
+      byPage: (settings?: PageSettings) => {
+        return this.getKeysPagingPage(options, settings);
       }
     };
   }
 
   private async *getKeysPagingPage(
-    options?: GetKeysOptionalParams
+    options?: GetKeysOptionalParams,
+    settings?: PageSettings
```

Comment:
`maxPageSize` is not implemented, right? should `{ continuationToken?: string }` be used instead of `PageSettings`?

Reply:
I think you're right that we shouldn't overpromise here.

Reply:
Revisiting this, it's a little tricky to use an anonymous type here since it gets referenced in multiple places. To avoid silently ignoring the parameter, I've added a runtime check to fail fast if someone passes it. A convenience client that wants to support it can still pull the parameter off and do the right thing before calling into the generated method.